python/ml4ir/base/model/calibration/temperature_scaling.py

```diff
@@ -344,11 +352,25 @@ def __init__(self, temperature, name='temperature', **kwargs):
         Parameters
         ----------
             temperature: float
-            temperature parameter to scale the input.
+                a float value to be used for initializing the temperature parameter to scale the
+                input.
         """
         super(TemperatureScalingLayer, self).__init__(name=name, **kwargs)
         self.temperature = temperature
 
+    def build(self, input_shape):
+        """initialize weights for temperature parameter
+        Parameters
+        ----------
+        input_shape: tuple
+            a tuple showing layer input shape
+        """
+        self.temperature_var = self.add_weight("temperature",
+                                               shape=[1],
+                                               dtype=tf.float32,
+                                               initializer=tf.keras.initializers.
+                                               Constant(value=self.temperature))
+
```

Comment:
defining temperature as a model parameter so that its value be saved/loaded using  `save()` and `load_model() ` tf functions